spikewrap/pipeline/load_data.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Dict, List, Union
+
+import spikeinterface.extractors as se
+
+from ..data_classes.preprocessing import PreprocessingData
+from ..utils import utils
+
+
+def load_data(
+    base_path: Union[Path, str],
+    sub_name: str,
+    sessions_and_runs: Dict[str, List[str]],
+    data_format: str = "spikeglx",
+) -> PreprocessingData:
+    """
+    Load raw data (in rawdata). If multiple runs are selected
+    in run_names, these will be stored as segments on a SpikeInterface
+    recording object.
+
+    Parameters
+    -----------
+
+    base_path : Union[Path, str]
+        Path where the rawdata folder containing subjects.
+
+    sub_name : str
+        Subject to preprocess. The subject top level dir should reside in
+        base_path/rawdata/ .
+
+    sessions_and_runs : Dict[str, Union[str, List]]
+        A dictionary containing the sessions and runs to run through the pipeline.
+        Each session should be a session-level folder name residing in the passed
+        `sub_name` folder. Each session to run is a key in the
+        `sessions_and_runs` dict.
+        For each session key, the value can be a single run name (str)
+        or a list of run names. The runs will be processed in the
+        order passed.
+
+    data_format : str
+        The data type format to load. Currently only "spikeglx" is accepted.
+
+    Returns
+    -------
+
+    PreprocessingData class containing SpikeInterface recording object and information
+    on the data filepaths.
+
+    TODO
+    ----
+    Figure out the format from the data itself, instead of passing as argument.
+    Do this when adding the next supported format.
+    """
+    empty_data_class = PreprocessingData(Path(base_path), sub_name, sessions_and_runs)
+
+    if data_format == "spikeglx":
```

Comment:
How many different loading there will be?
If they will be just two case we could have switch like:
```python
if data_format == "spikeglx":
    loaded_data = _load_spikeglx_data(empty_data_class)
elif data_format == "something_else":
    loaded_data = _load_something_else_data(empty_data_class)
else:
    raise RuntimeError("`data_format` not recognised.")
```

Reply:
Great point we may support 5+ file formats so an important consideration, opened #128 and can consider further when implementing OpenEphys support.